frame/support/src/storage/types/key.rs

```diff
@@ -196,6 +196,9 @@ impl_encode_like_tuples!(A, B, C, D, E, F, G, H, I, J, K, L, M, O, P);
 impl_encode_like_tuples!(A, B, C, D, E, F, G, H, I, J, K, L, M, O, P, Q);
 impl_encode_like_tuples!(A, B, C, D, E, F, G, H, I, J, K, L, M, O, P, Q, R);
 
+impl<'a, T: EncodeLike<U>, U: Encode> EncodeLikeTuple<U> for codec::Ref<'a, T, U> {}
```

Comment:
The name of this trait should already give you a hint that you're only supposed to implement it on tuples. Why is this being implemented on `U`?

Reply:
I am trying to implement your suggestion https://github.com/paritytech/substrate/pull/10621#discussion_r985419556, for it is needed implement this trait also for codec::Ref

Reply:
Because `contains_key` method which I am trying to invoke here have the following trait bounds
`pub fn contains_key<KArg: EncodeLikeTuple<Key::KArg> + TupleToEncodedIter>(key: KArg) -> bool`
https://github.com/Mr-Leshiy/substrate/blob/09583a983add3fc443a408ca0670e6112d570905/frame/support/src/storage/types/nmap.rs#L125

Reply:
The same for other methods where it is relevant to replace it with `Ref::from(&key)`

Reply:
This is really not the correct way to do so. See the comment on `EncodeLikeTuple`:

```rust
/// Marker trait to indicate that each element in the tuple encodes like the corresponding element
/// in another tuple.
///
/// This trait is sealed.
pub trait EncodeLikeTuple<T>: crate::storage::private::Sealed {}
```

There's no guarantee that T or U in your implementation here are tuples. If you want to implement `EncodeLikeTuple` for `codec::Ref`, then you can refer to the `impl_encode_like_tuples` macro, or even add additional implementations inside of it.